modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java

```diff
@@ -130,10 +130,10 @@ public void processOpts() {
     @SuppressWarnings({ "static-method", "unchecked" })
     public Map<String, Object> postProcessAllModels(Map<String, Object> objs) {
         if (supportsInheritance) {
-            // Index all CodegenModels by name.
+            // Index all CodegenModels by model name.
             Map<String, CodegenModel> allModels = new HashMap<String, CodegenModel>();
             for (Entry<String, Object> entry : objs.entrySet()) {
-                String modelName = entry.getKey();
+                String modelName = toModelName(entry.getKey());
```

Comment:
The change was within a block of code that I myself added in a previous PR and as far as I am aware is only used by the JavaScript code generator because that's the only one that sets `DefaultCodegen.supportsInheritance = true` **and** knows about the `parentModel` and `interfaceModel` fields I added to `CodegenModel`. Basically what we're doing here is indexing all the `CodegenModel`s by their **codegen model** name (which starts with an uppercase letter) so that we can populate those new fields. The previous code was incorrect, as it was keying the map on **Swagger model** name, which _could_ start with a lowercase letter if that's how the type was defined in the YAML. If you look at the way `allModels` is used in the next code block you'll see that it's being used to resolve the parent and interface **codegen model** names (not the **Swagger model** names) to the corresponding codegen models in order to populate the new fields. I had to make this `toModelName(entry.getKey())` change because previously it would only work if the Swagger model definition name was in uppercase.